src/nvrtc/safe.rs

```diff
@@ -222,6 +222,51 @@ impl CompileOptions {
     }
 }
 
+pub struct RustPtx {
+    kernel_path: PathBuf,
+}
+
+use std::process::Command;
+impl RustPtx {
+    pub fn new(kernel_path: PathBuf) -> RustPtx {
+        RustPtx { kernel_path }
+    }
+
+    pub fn build_ptx(&self) -> Result<PathBuf, String> {
```

Comment:
What are your thoughts on turning this into a single function that returns a Ptx object? I'm thinking interface like this:
```rust
pub fn compile_rust_to_ptx<S: AsRef<Path>>(kernel_path: S) -> Ptx {
    ... //current code
    Ptx::from_file(ptx_path)
}
```

Reply:
yep, just finished up a PtxCrate struct that has a TryFrom<PathBuf> and methods ptx_files, clean, and build_ptx

Reply:
PtxCrate::compile_crate_to_ptx is now a-go